beetmoverscript/src/beetmoverscript/script.py

```diff
@@ -198,11 +200,29 @@ async def push_to_maven(context):
     context.checksums = dict()  # Needed by downstream calls
     context.raw_balrog_manifest = dict()  # Needed by downstream calls
 
+    check_maven_artifact_map(context)
+
     # overwrite artifacts_to_beetmove with the declarative artifacts ones
     context.artifacts_to_beetmove = task.get_upstream_artifacts(context, preserve_full_paths=True)
     await move_beets(context, context.artifacts_to_beetmove, artifact_map=context.task["payload"]["artifactMap"])
 
 
+def check_maven_artifact_map(context):
+    """Check that versions in artifact map to be valid Maven versions"""
+    version = context.task["payload"]["version"]
+    try:
+        MavenVersion.parse(version)
+    except PatternNotMatchedError:
+        raise ScriptWorkerTaskException(f"bad version '{version}'")
```

Comment:
Nit: I usually like when error messages are super explicit. It helps to know what happened whenever the exception is raised in prod and you have no easy way to reproduce the problem. Here, I think we can extend the message to something like: 

```suggestion
        raise ScriptWorkerTaskException(f"Version defined in the payload does not match the pattern of a MavenVersion. Got: {version}")
```

We can even go one step further by re-raising the original error. This way, we know what didn't MavenVersion didn't like.

```suggestion
        raise ScriptWorkerTaskException(f"Version defined in the payload does not match the pattern of a MavenVersion. Got: {version}") from e
```

What do you think?

Reply:
Great! I had never used the `raise from` statement even though I knew it existed.